feat(aws/elastigroup) : Added support for `logging` in Elastigroup-AWS.

sharadkesarwani opened the pull request:
feat(aws/elastigroup) : Added support for `logging` in Elastigroup-AWS.

# Jira Ticket
Ref: https://spotinst.atlassian.net/browse/SPOTAUT-17203

sharadkesarwani committed:
Added support for logging in Elastigroup-AWS.

Added support for logging in Elastigroup-AWS.

sharadkesarwani committed:
Added test for ElastigroupAWS_Logging

Added test for ElastigroupAWS_Logging

sharadkesarwani committed:
Merge branch 'main' into SPOTAUT-17203-support-logging-eg-aws

sharadkesarwani committed:
Upgraded sdk version and updated change log

Upgraded sdk version and updated change log

chandra1-n commented on docs/resources/elastigroup_aws.md:
```diff
@@ -134,6 +134,14 @@ resource "spotinst_elastigroup_aws" "default-elastigroup" {
 
+  logging {
+      export {
+        s3      {
```
please check the alignment

sharadkesarwani replied:
fixed alignment

chandra1-n commented on spotinst/resource_spotinst_elastigroup_aws_test.go:
```diff
@@ -4724,3 +4724,64 @@ const testAwaitCapacity_EmptyFields = `
 `
 
 // endregion
+// region OceanAWS: Baseline
```
modify the commented line according to the test name

sharadkesarwani replied:
updated to region Elastigroup: Logging

chandra1-n commented on spotinst/resource_spotinst_elastigroup_aws_test.go:
```diff
@@ -4724,3 +4724,64 @@ const testAwaitCapacity_EmptyFields = `
+				ResourceName: resourceName,
+				Config: createElastigroupTerraform(&GroupConfigMetadata{
+					groupName: groupName,
+					/*launchConfig: testElastigroupLoggingAWSConfig_Create,*/
```
what is the reason launchConfig is commented

sharadkesarwani replied:
its not required, removed commented code

chandra1-n commented on spotinst/resource_spotinst_elastigroup_aws_test.go:
```diff
@@ -4724,3 +4724,64 @@ const testAwaitCapacity_EmptyFields = `
+				ResourceName: resourceName,
+				Config: createElastigroupTerraform(&GroupConfigMetadata{
+					groupName: groupName,
+					/*launchConfig: testElastigroupLoggingAWSConfig_EmptyFields,*/
```
please remove if not used

sharadkesarwani replied:
removed commented code

chandra1-n commented on spotinst/resource_spotinst_elastigroup_aws_test.go:
```diff
@@ -4724,3 +4724,64 @@ const testAwaitCapacity_EmptyFields = `
+  logging {
+    export {
+      s3 { 
+		id = "di-5fae075b"
```
please check the alignment

sharadkesarwani replied:
fixed alignment

chandra1-n commented on spotinst/resource_spotinst_elastigroup_aws_test.go:
```diff
@@ -4724,3 +4724,64 @@ const testAwaitCapacity_EmptyFields = `
+		},
+	})
+}
+
```
update test is missing

sharadkesarwani replied:
Added update test

chandra1-n reviewed (changes requested):
please fix the review comments

sharadkesarwani committed:
Updated test(TestAccSpotinstElastigroupAWS_Logging) and formatting in documentation.

Updated test(TestAccSpotinstElastigroupAWS_Logging) and formatting in documentation.
